traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -103,62 +89,136 @@ func UpdateHandler(w http.ResponseWriter, r *http.Request) {
-		return !strings.HasPrefix(strings.ToLower(s), "f")
+		return !strings.HasPrefix(s, "f")
 	}
 
-	updatedPtr := (*bool)(nil)
 	if hasUpdated {
 		updatedBool := strToBool(updated)
-		updatedPtr = &updatedBool
+		if updatedBool {
+			if err = dbhelpers.QueueUpdateForServer(inf.Tx.Tx, int64(serverID)); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting update status: "+err.Error()))
+				return
+			}
+		} else {
+			if err = dbhelpers.SetApplyUpdateForServer(inf.Tx.Tx, int64(serverID)); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting update status: "+err.Error()))
+				return
+			}
+		}
 	}
-	revalUpdatedPtr := (*bool)(nil)
+
 	if hasRevalUpdated {
 		revalUpdatedBool := strToBool(revalUpdated)
-		revalUpdatedPtr = &revalUpdatedBool
+		if revalUpdatedBool {
+			if err = dbhelpers.QueueRevalForServer(inf.Tx.Tx, int64(serverID)); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting reval status: "+err.Error()))
+				return
+			}
+		} else {
+			if err = dbhelpers.SetApplyRevalForServer(inf.Tx.Tx, int64(serverID)); err != nil {
+				api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting update status: "+err.Error()))
+				return
+			}
+		}
 	}
 
-	if err := setUpdateStatuses(inf.Tx.Tx, hostName, updatedPtr, revalUpdatedPtr); err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting updated statuses: "+err.Error()))
-		return
+	if hasConfig_update_time {
+		if err = dbhelpers.QueueUpdateForServerWithTime(inf.Tx.Tx, int64(serverID), configUpdateTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting config update time: "+err.Error()))
+			return
+		}
 	}
 
+	if hasConfig_apply_time {
+		if err = dbhelpers.SetApplyUpdateForServerWithTime(inf.Tx.Tx, int64(serverID), configApplyTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting config apply time: "+err.Error()))
+			return
+		}
+	}
+
+	if hasRevalidate_update_time {
+		if err = dbhelpers.QueueRevalForServerWithTime(inf.Tx.Tx, int64(serverID), revalUpdateTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting reval update time: "+err.Error()))
+			return
+		}
+	}
+
+	if hasConfig_apply_time {
+		if err = dbhelpers.SetApplyUpdateForServerWithTime(inf.Tx.Tx, int64(serverID), revalApplyTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("setting reval apply time: "+err.Error()))
+			return
+		}
+	}
+
+	hostName, _, _ := dbhelpers.GetServerNameFromID(inf.Tx.Tx, serverID)
+
 	respMsg := "successfully set server '" + hostName + "'"
 	if hasUpdated {
-		respMsg += " updated=" + strconv.FormatBool(strToBool(updated))
+		respMsg += " updated=" + updated
```

Comment:
We should show users  what was actually done, not merely what they requested be done.

Reply:
If you mean we should tell them that we actually set the `*_update_time` values rather than the bools, I could see that. I'm not sure at what point you expose your business logic externally in requests. Also I was unsure if clients were relying on the original message I did not want to change it.

If someone is still using a v2 or v3 api, the response they get will be the same. If they take advantage of the new timestamp parameters, then they will be informed of that.